core/src/main/java/org/apache/iceberg/avro/Codec.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.avro;
+
+public enum Codec {
```

Comment:
why do we want a new class for Avro codec? I don't see it used anywhere. We prefer to not introduce new public classes, as we have to maintain them whereas private and protected classes and methods can be changed easily.

Reply:
It seemed more readable to me to have it separate from the main Avro class, but you are right , what's more important is that we do not expose the class as it forces maintenance requirements on us going into the future.

Reply:
Yes, I would leave this class as a private enum within Avro as it was.